Failure to compile on Alpine Linux

# Defect Report

## Issue type

- [x] Defect - Compilation or `make check` issue - Attach `config.log` and provide relevant system information such as `lscpu` - Make sure dependencies are installed

## System information

- OS Name: Alpine Linux
- OS Version: 3.17
- CPU: amd64

## Aircrack-ng version

- Version: 1.7
- Commit Revision hash: [c0645a3](https://github.com/aircrack-ng/aircrack-ng/commit/c0645a301bf0bc3587a056a5e5c6024fb6312b81)

## Airmon-ng debug information

N/A

## Defect

### Details

The Alpine GitHub action fails to compile on [amd64](https://github.com/aircrack-ng/aircrack-ng/actions/runs/4070723991/jobs/7011849213#step:4:1371):

```
 > [linux/amd64 builder 7/7] RUN set -x 	&& make distclean || : && 		autoreconf -vif && 		set -e; 			./configure --with-experimental --with-ext-scripts --enable-maintainer-mode --without-opt --prefix=/usr && 			make -j3 && 		set +e && 			if ! make check -j3; then 				echo "Processor: $(uname -m)"; 				for file in `grep -l "(exit status: [1-9]" test/*.log`; do 					echo "[*] Test ${file}:"; 					cat "${file}"; 				done; 				exit 1; 			fi && 		set -e && 			make install DESTDIR=/output:
#17 25.28    63 |         u_short ether_type;
#17 25.28       |         ^~~~~~~
#17 25.29 /aircrack-ng/include/aircrack-ng/third-party/ethernet.h:70:9: error: unknown type name 'u_char'
#17 25.29    70 |         u_char octet[ETHER_ADDR_LEN];
#17 25.29       |         ^~~~~~
#17 25.32 cc1: note: unrecognized command-line option '-Wno-gnu-zero-variadic-macro-arguments' may have been intended to silence earlier diagnostics
#17 25.33 make[2]: *** [Makefile:5124: src/packetforge-ng/packetforge_ng-packetforge-ng.o] Error 1
#17 25.33 make[2]: *** Waiting for unfinished jobs....
#17 25.47 make[1]: *** [Makefile:5535: all-recursive] Error 1
#17 25.47 make: *** [Makefile:2917: all] Error 2
```

### How to reproduce the issue

Commit in the repo

## Related issues

#2418 might possibly be related as `u_short` is in `sys/types.h` IIRC.

--- a/src/airbase-ng/airbase-ng.c
+++ b/src/airbase-ng/airbase-ng.c
@@ -38,7 +38,12 @@
 #include "config.h"
 #endif
 
+#ifdef linux
+#include <linux/rtc.h>
+#endif
+
 #include <sys/types.h>
+#include <sys/ioctl.h>
 #include <sys/time.h>
 
 #include <pthread.h>

--- a/src/aireplay-ng/aireplay-ng.c
+++ b/src/aireplay-ng/aireplay-ng.c
@@ -39,8 +39,13 @@
 #include "config.h"
 #endif
 
+#if defined(linux)
+#include <linux/rtc.h>
+#endif
+
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <sys/ioctl.h>
 #include <sys/time.h>
 
 #include <netinet/in.h>

--- a/src/airtun-ng/airtun-ng.c
+++ b/src/airtun-ng/airtun-ng.c
@@ -38,6 +38,11 @@
 #include "config.h"
 #endif
 
+#ifdef linux
+#include <linux/rtc.h>
+#endif
+
+#include <sys/ioctl.h>
 #include <sys/time.h>
 
 #include <unistd.h>
@@ -46,6 +51,8 @@
 #include <stdio.h>
 #include <errno.h>
 #include <getopt.h>
+
+#include <fcntl.h>
 
 #include "aircrack-ng/defs.h"
 #include "aircrack-ng/version.h"

--- a/src/easside-ng/easside-ng.c
+++ b/src/easside-ng/easside-ng.c
@@ -29,6 +29,7 @@
 #include <err.h>
 #include <fcntl.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <sys/stat.h>
 #include <signal.h>
 #include <sys/time.h>

--- a/src/packetforge-ng/packetforge-ng.c
+++ b/src/packetforge-ng/packetforge-ng.c
@@ -43,6 +43,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/time.h>
+#include <sys/types.h>
 #include <getopt.h>
 
 #include "aircrack-ng/defs.h"

--- a/src/tkiptun-ng/tkiptun-ng.c
+++ b/src/tkiptun-ng/tkiptun-ng.c
@@ -38,9 +38,21 @@
 #include "config.h"
 #endif
 
+#if defined(linux)
+#include <linux/rtc.h>
+#endif
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <sys/wait.h>
 #include <sys/time.h>
 
+#include <netinet/in.h>
+#include <arpa/inet.h>
 #include <unistd.h>
+#include <dirent.h>
+#include <signal.h>
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -49,8 +61,15 @@
 #include <getopt.h>
 
 #include <fcntl.h>
+#include <ctype.h>
 
 #include <limits.h>
+
+#if defined(linux)
+#include <netinet/in_systm.h>
+#include <netinet/ip.h>
+#include <netinet/tcp.h>
+#endif
 
 #include "aircrack-ng/defs.h"
 #include "aircrack-ng/version.h"

